connector/src/main/java/io/pixelsdb/pixels/trino/PixelsMetadata.java

```diff
@@ -401,6 +417,58 @@ else if (scheme != storageScheme)
         {
             throw new TrinoException(PixelsErrorCode.PIXELS_METADATA_ERROR, e);
         }
+        
+        if(pkColNames != null && !pkColNames.isEmpty())
+        {
+            try
+            {
+                MetadataService metadataService = this.metadataProxy.getMetadataService();
+                io.pixelsdb.pixels.common.metadata.domain.Table table = metadataService.getTable( schemaName, tableName);
+                List<Column> pixelsColumns = metadataService.getColumns(schemaName, tableName, false);
+                Layout layout = metadataService.getLatestLayout(schemaName, tableName);
+                String keyColumnsString = getKeyColumnsString(pkColNames, pixelsColumns, schemaTableName);
+                MetadataProto.SinglePointIndex.Builder singlePointIndexbuilder = MetadataProto.SinglePointIndex.newBuilder();
+                singlePointIndexbuilder
+                        .setKeyColumns(keyColumnsString)
+                        .setPrimary(true)
+                        .setUnique(true)
+                        .setIndexScheme(pkScheme)
+                        .setTableId(table.getId())
+                        .setSchemaVersionId(layout.getSchemaVersionId());
```

Comment:
layout.schemaVersionId has not been merged into the master branch of pixels. Remove this line and add it back after retina-dev is merged into master.